app/controllers/submissions_controller.rb

```diff
@@ -213,7 +213,7 @@ def collect_submissions
                                                   collect_current: collect_current,
                                                   apply_late_penalty: apply_late_penalty,
                                                   notify_socket: true)
-      session[:job_id] = @current_job.job_id
+      CollectSubmissionsChannel.broadcast_to(@current_user, ActiveJob::Status.get(@current_job).to_h)
```

Comment:
Make `current_job` a local variable rather than an instance variable (since it no longer needs to be accessed from a rendered view).